scheduler/filter/filter.go

```diff
@@ -33,6 +33,7 @@ func init() {
 		&SlotsFilter{},
 		&DependencyFilter{},
 		&AffinityFilter{},
+		&WhitelistFilter{},
 		&ConstraintFilter{},
```

Comment:
Are you sure no more changes are needed in `filter.go`? I'm looking more but notice that filters are applied serially, since each filter only serves to further reduce the number of available nodes. The whitelist filter has a fundamentally opposite purpose, so it should be applied last, and the final set should be a union of the result of old filters with whitelisted nodes.